cohortextractor2/backends/base.py

```diff
@@ -25,48 +43,78 @@ def __init_subclass__(cls, **kwargs):
 
         # Register each Backend by its id so we can identify it from an environment variable
         register_backend(cls)
+
         # Make sure each Backend knows what its tables are
         cls.tables = set()
         for name, value in vars(cls).items():
             if isinstance(value, SQLTable):
-                cls._init_table(name, value)
+                cls._init_table(value)
                 cls.tables.add(name)
 
-    @classmethod
-    def _init_table(cls, name, table):
-        table.learn_patient_join(cls.patient_join_column)
-        # Validate that the table correctly implements the contract it claims to, if any
-        contract = table.implements
-        if contract:
-            contract.validate_implementation(cls, name, table)
-
     def __init__(self, database_url, temporary_database=None):
         self.database_url = database_url
         self.temporary_database = temporary_database
 
+    @classmethod
+    def _init_table(cls, table: SQLTable) -> None:
+        """
+        Initialises the table
+
+        Args:
+            name: Name of Table
+            table: SQLTable Object
+
+        Returns:
+            None
+        """
```

Comment:
I'd argue that the type annotations do a better job of documenting this method than the docstring -- and they can't get out of date.

Reply:
agree- esp with the doc work that George is doing already.